includes/base/widget-base.php

```diff
@@ -5,14 +5,56 @@
 	exit; // Exit if accessed directly.
 }
 
+/**
+ * Widget Base
+ *
+ * Base class extended to register Elementor widgets.
+ *
+ * This class must be extended for each widget.
+ *
+ * @abstract
+ */
 abstract class Widget_Base extends Element_Base {
 
+	/**
+	 * Whether the widget has content.
+	 *
+	 * Used in cased where the widget has no content. When widgets uses only
+	 * skins to display dynamic content generated on the server. For example the
+	 * posts widget in Elemenrot Pro. Default is true, the widget has content
+	 * template.
+	 *
+	 * @access protected
+	 *
+	 * @var bool
+	 */
 	protected $_has_template_content = true;
 
+	/**
+	 * Retrieve element type.
+	 *
+	 * Get the element type, in this case `widget`.
+	 *
+	 * @access public
+	 * @static
+	 *
+	 * @return string Control type.
+	 */
 	public static function get_type() {
 		return 'widget';
 	}
 
+	/**
+	 * Retrieve default edit tools.
+	 *
+	 * Get the default edit tools of the widget. This method is used to set
+	 * initial tools - by default widgets have the duplicate and removal tools.
```

Comment:
the following tools: Edit, Duplicate, Save, Remove.

Reply:
@matipojo This method returns only duplicate and removal tools. See the code.

Reply:
Get the default edit tools of the widget. This method is used to set initial tools - it adds duplicate and Remove tools on top of Edit and Save tools.